internal/resolution/variable_sources/olm/olm.go

```diff
@@ -14,24 +14,36 @@ import (
 var _ input.VariableSource = &OLMVariableSource{}
 
 type OLMVariableSource struct {
-	packageNames []string
+	operators []operatorsv1alpha1.Operator
 }
 
-func NewOLMVariableSource(packageNames ...string) *OLMVariableSource {
+func NewOLMVariableSource(operators ...operatorsv1alpha1.Operator) *OLMVariableSource {
 	return &OLMVariableSource{
-		packageNames: packageNames,
+		operators: operators,
 	}
 }
 
 func (o *OLMVariableSource) GetVariables(ctx context.Context, entitySource input.EntitySource) ([]deppy.Variable, error) {
 	var inputVariableSources []input.VariableSource
 
 	// build required package variable sources
-	for _, packageName := range o.packageNames {
-		inputVariableSources = append(inputVariableSources, required_package.NewRequiredPackage(packageName))
+	for _, operator := range o.operators {
+		rps, err := o.requiredPackageFromOperator(&operator)
+		if err != nil {
+			return nil, err
+		}
+		inputVariableSources = append(inputVariableSources, rps)
 	}
 
 	// build variable source pipeline
 	variableSource := crd_constraints.NewCRDUniquenessConstraintsVariableSource(bundles_and_dependencies.NewBundlesAndDepsVariableSource(inputVariableSources...))
 	return variableSource.GetVariables(ctx, entitySource)
 }
+
+func (o *OLMVariableSource) requiredPackageFromOperator(operator *operatorsv1alpha1.Operator) (*required_package.RequiredPackageVariableSource, error) {
+	var opts []required_package.RequiredPackageOption
+	if operator.Spec.Version != "" {
+		opts = append(opts, required_package.InVersionRange(operator.Spec.Version))
```

Comment:
@perdasilva given the direction set out in #162, we're saying that only individual versions will be allowed in the API. However, an individual version is a valid range, and I am still of the belief that we'll want to give users the ability to specify a range at some point.

So my advice is:
1. Let's make the `spec.version` regex be a regex that permits only individual versions
2. Let's keep `InVersionRange` as the under-the-hood implementation of the constraint.

Reply:
Yeah, I agree with everything you said. The original regex was for a single version (got my wires crossed - managed to get a pretty decent one for ranges, but I don't think there's a regex that matches all cases - it could be that we'll need to match *most* but still verify under the hood. ChatGPT ftw XD